src/api/spec/models/obs_factory/staging_project_spec.rb

```diff
@@ -39,4 +53,623 @@
       it { expect(subject.adi_staging?).to be false }
     end
   end
+
+  describe 'self.for' do
+    context 'openSUSE:Factory' do
+      subject { ObsFactory::StagingProject.for(factory_distribution) }
+
+      it { expect(subject.count).to eq(3) }
+    end
+
+    context 'openSUSE:Factory:PowerPC' do
+      let(:factoryppc_distribution) { ObsFactory::Distribution.find(factoryppc.name) }
+      subject { ObsFactory::StagingProject.for(factoryppc_distribution) }
+
+      it { expect(subject.count).to eq(3) }
+    end
+
+    context 'openSUSE:13.2' do
+      let(:opensuse_132_distribution) { ObsFactory::Distribution.find(opensuse_132.name) }
+      subject { ObsFactory::StagingProject.for(opensuse_132_distribution) }
+
+      it { expect(subject.count).to eq(2) }
+    end
+  end
+
+  describe '.name' do
+    subject { staging_project_a }
+
+    it { expect(subject.name).to eq('openSUSE:Factory:Staging:A') }
```

Comment:
I had to scroll up and down a couple of times to understand this test.

Please change it to something like this:

```ruby
subject { create(:obs_factory_staging_project, name: 'openSUSE:Factory:Staging:A') }

it { expect(subject.name).to eq('openSUSE:Factory:Staging:A') }
```

Reply:
This would also come handy for many of the other tests below